src/game-loop/src/components/generic/ItemCarrierComponent.cpp

```diff
@@ -245,3 +236,35 @@ std::vector<ItemType> ItemCarrierComponent::get_items() const
 
     return out;
 }
+
+void ItemCarrierComponent::place_passive_item(entt::entity &slot, entt::entity item)
+{
+    auto& registry = EntityRegistry::instance().get_registry();
+
+    if (slot != entt::null)
+    {
+        auto& item_component = registry.get<ItemComponent>(slot);
+        notify({item, ItemCarrierEvent::EventType::REMOVED, item_component.get_type()});
+        registry.destroy(slot);
+    }
+
+    slot = item;
+}
+
+void ItemCarrierComponent::recalculate_modifiers()
+{
+    const auto items = get_items();
+
+    _modifiers.can_climb_vertical_surfaces = has_item(ItemType::GLOVE, items);
+    _modifiers.additional_jump_on_top_damage = has_item(ItemType::SPIKE_SHOES, items) ? 1 : 0;
+    _modifiers.additional_throw_x_velocity = has_item(ItemType::MITT, items) ? 0.2f : 0.0f;
+    _modifiers.additional_jump_y_velocity = has_item(ItemType::SPRING_SHOES, items) ? 0.05f : 0.0f;
+}
+
+bool ItemCarrierComponent::has_item(ItemType type, const std::vector<ItemType>& items) const
```

Comment:
Similar method introduced in https://github.com/dbeef/spelunky-psp/pull/97

May be redundant